core/jobs/types/model_property.py

```diff
@@ -76,15 +92,20 @@ def model_kind(self):
         return self._model_kind
 
     @property
-    def property_name(self):
+    def property_name(self) -> str:
         """Returns the name of the property this instance refers to.
 
         Returns:
             str. The name of the property.
         """
         return self._property_name
 
-    def yield_value_from_model(self, model):
+    # This method yields the values of properties of a model and that values
+    # can be of type string, list, integer and other types too. So, that's
+    # why Iterator[Any] type is used as a yield value of function.
+    def yield_value_from_model(
+        self, model: base_models.BaseModel
+    ) -> Iterator[Any]:
```

Comment:
Value of property of modal can be of type string, int, list, and other types too. That's why i added Iteratr[Any] here.

Reply:
makes sense